app/discord.server.ts

```diff
@@ -0,0 +1,46 @@
+import crypto from "node:crypto";
+import DiscordOauth2 from "discord-oauth2";
+
+const DISCORD_REDIRECT_URI =
+  process.env.NODE_ENV === "production"
+    ? "https://opensourcehub.io/verify-discord"
+    : "http://localhost:3000/verify-discord";
+
+const DISCORD_SCOPES = ["identify"];
+
+export async function getDiscordAccessToken(code: string) {
+  const discordAuth = new DiscordOauth2();
+
+  const token = await discordAuth.tokenRequest({
+    clientId: process.env.DISCORD_CLIENT_ID,
+    clientSecret: process.env.DISCORD_CLIENT_SECRET,
```

Comment:
**Question (maybe blocking)**: How are we keeping these values secret?

Reply:
They're pulled from a `.env` file that isn't checked into source control. These variables will only be available on production and, because all this code happens on the server, they won't be exposed to any client. I do need to fail gracefully for local development, though.